client/src/main/java/org/eclipse/hono/client/impl/HonoConnectionImpl.java

```diff
@@ -769,9 +832,20 @@ public void closeAndFree(
 
                     } else if (HonoProtonHelper.isLinkEstablished(sender)) {
 
-                        log.debug("sender open [target: {}, sendQueueFull: {}]", targetAddress, sender.sendQueueFull());
-                        // wait on credits a little time, if not already given
-                        if (sender.getCredit() <= 0) {
+                        log.debug("sender open [target: {}, sendQueueFull: {}, remote max-message-size: {}]",
+                                targetAddress, sender.sendQueueFull(), sender.getRemoteMaxMessageSize());
+                        final long remoteMaxMessageSize = Optional.ofNullable(sender.getRemoteMaxMessageSize())
+                                .map(UnsignedLong::longValue)
+                                .orElse(0L);
+                        if (remoteMaxMessageSize < clientConfigProperties.getMinMessageSize()) {
+                            // peer won't accept our (biggest) messages
+                            sender.close();
+                            log.debug("peer does not support minimum message size [required: {}, supported: {}",
+                                    clientConfigProperties.getMinMessageSize(), remoteMaxMessageSize);
+                            senderPromise.tryFail(new ClientErrorException(HttpURLConnection.HTTP_PRECON_FAILED,
+                                    "peer does not meet sender's minimum max-message-size requirement"));
```

Comment:
I think throwing a `ClientErrorException` here means that we have to adapt the code for sending telemetry messages for example by catching this `ClientErrorException` and rethrowing it back as a ServerErrorException, so that the device gets a 5xx error here.

Reply:
Well, both the `DownstreamSender.send()` and `HonoConnection.createSender()` JavaDoc allow the returned Future to be failed with a ClientErrorException. Or am I missing something?

Reply:
Yes, I see there is a case already where a ClientErrorException is thrown in `HonoConnection.createSender()`.
So that is an issue independent of this PR really, preventing a ClientErrorException on `createSender()` for a telemetry/event sender to be propagated to the device. I'll create an issue for that.